### PR TITLE
feat: add an option to enable pprof profiling

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
- Adds an option to enable pprof profiling

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Added an option to enable pprof profiling with --enable-pprof. The profiling server port can be configured using --http-endpoint.
```

### CODE REVIEW

File: cmd/csi-provisioner/csi-provisioner.go [hunk 1]
```diff
@@ -83,8 +84,9 @@ var (
...
-	httpEndpoint            = flag.String("http-endpoint", "", "The TCP network address where the HTTP server for diagnostics, including metrics and leader election health check, will listen (example: `:8080`). The default is empty string, which means the server is disabled. Only one of `--metrics-address` and `--http-endpoint` can be set.")
+	httpEndpoint            = flag.String("http-endpoint", "", "The TCP network address where the HTTP server for diagnostics, including pprof, metrics and leader election health check, will listen (example: `:8080`). The default is empty string, which means the server is disabled. Only one of `--metrics-address` and `--http-endpoint` can be set.")
 	metricsPath             = flag.String("metrics-path", "/metrics", "The HTTP path where prometheus metrics will be exposed. Default is `/metrics`.")
+	enableProfile           = flag.Bool("enable-pprof", false, "enable pprof profiling")
```
Comment (ggriffiths) [inline cmd/csi-provisioner/csi-provisioner.go:89]: Can you elaborate on the description here? I like the description you gave in readme.md:
`Enable pprof profiling on the TCP network address specified by --http-endpoint`

**Review by ggriffiths — COMMENTED** [review]
I don't see any issues with adding this. Mostly looks good to me! Just one nit on the flag description

File: cmd/csi-provisioner/csi-provisioner.go (hunk 1 above)
Comment (aramase) [reply cmd/csi-provisioner/csi-provisioner.go]: Thanks for the review @ggriffiths! Updated it.

File: cmd/csi-provisioner/csi-provisioner.go [hunk 2]
```diff
@@ -83,8 +84,9 @@ var (
...
-	httpEndpoint            = flag.String("http-endpoint", "", "The TCP network address where the HTTP server for diagnostics, including metrics and leader election health check, will listen (example: `:8080`). The default is empty string, which means the server is disabled. Only one of `--metrics-address` and `--http-endpoint` can be set.")
+	httpEndpoint            = flag.String("http-endpoint", "", "The TCP network address where the HTTP server for diagnostics, including pprof, metrics and leader election health check, will listen (example: `:8080`). The default is empty string, which means the server is disabled. Only one of `--metrics-address` and `--http-endpoint` can be set.")
 	metricsPath             = flag.String("metrics-path", "/metrics", "The HTTP path where prometheus metrics will be exposed. Default is `/metrics`.")
+	enableProfile           = flag.Bool("enable-pprof", false, "Enable pprof profiling on the TCP network address specified by --http-endpoint")
```
Comment (jsafrane) [inline cmd/csi-provisioner/csi-provisioner.go:89]: Please add the URL where pprof is available. Is it `/debug/pprof/`?

File: README.md [hunk 3]
```diff
@@ -74,6 +74,8 @@ Note that the external-provisioner does not scale with more replicas. Only one e
 
 * `--metrics-path`: The HTTP path where prometheus metrics will be exposed. Default is `/metrics`.
 
+* `--enable-pprof`: Enable pprof profiling on the TCP network address specified by `--http-endpoint`.
```
Comment (jsafrane) [inline README.md:77]: Again, please add directory where the pprof is available.

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (jsafrane) [inline .vscode/settings.json:1]: Please do not add vscode metadata to git.
(feel free to update `.gitignore`)

File: README.md [hunk 5]
```diff
@@ -9,23 +9,23 @@ The external-provisioner is an external controller that monitors `PersistentVolu
 
 This information reflects the head of this branch.
 
-| Compatible with CSI Version | Container Image | [Min K8s Version](https://kubernetes-csi.github.io/docs/kubernetes-compatibility.html#minimum-version) | [Recommended K8s Version](https://kubernetes-csi.github.io/docs/kubernetes-compatibility.html#recommended-version) |
-| ------------------------------------------------------------------------------------------ | -------------------------------| --------------- | ------------- |
-| [CSI Spec v1.5.0](https://github.com/container-storage-interface/spec/releases/tag/v1.5.0) | k8s.gcr.io/sig-storage/csi-provisioner | 1.20 | 1.22 |
+| Compatible with CSI Version                                                                | Container Image                        | [Min K8s Version](https://kubernetes-csi.github.io/docs/kubernetes-compatibility.html#minimum-version) | [Recommended K8s Version](https://kubernetes-csi.github.io/docs/kubernetes-compatibility.html#recommended-version) |
```
Comment (jsafrane) [inline README.md:12]: We had whitespaces and nice tables sometime ago, but they're hard to maintain - image names and URLs have different length and I think it's better not to have the whitespaces there. If you insist, create at least a separate PR.

Comment (aramase) [reply README.md]: No strong preference here, the vscode plugins I've installed alter it. Reverted those changes.

**Pull request merged**